app/src/main/AndroidManifest.xml

```diff
@@ -1,10 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     package="com.example.filemanager">
 
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```

Comment:
I believe the `READ_EXTERNAL_STORAGE` is included in the `WRITE_EXTERNAL_STORAGE`. I guess you can remove it.

Reply:
I read from multiple sources that we must include both READ and WRITE

Reply:
If you declare the `WRITE_EXTERNAL_STORAGE` then the `READ_EXTERNAL_STORAGE` is redundant. If you have the rights to write then you also can read by default.